libraries/AP_RCProtocol/AP_RCProtocol_IBUS.cpp

```diff
@@ -20,10 +20,12 @@
 
 #include "AP_RCProtocol_IBUS.h"
 
+static int ia6_hack = 0;
+
 // decode a full IBUS frame
-bool AP_RCProtocol_IBUS::ibus_decode(const uint8_t frame[IBUS_FRAME_SIZE], uint16_t *values, bool *ibus_failsafe)
+bool AP_RCProtocol_IBUS::ibus_decode_std(const uint8_t frame[IBUS_FRAME_SIZE], uint16_t *values, bool *ibus_failsafe)
 {
-    uint32_t chksum = 96;
+    uint16_t chksum = 96;
```

Comment:
Are you sure this is correct?

ie. the checksum definition *must* state that your result is modulo UINT16_MAX.  If it does not then you may pass the checksum when you shouldn't.